platform/security/core/security-core-impl/src/test/java/ddf/security/assertion/impl/SecurityAssertionImplTest.java

```diff
@@ -72,42 +81,175 @@ public static Document readXml(InputStream is)
+                        .size());
         assertNull(assertion.getPrincipal());
         assertNull(assertion.getNotBefore());
         assertNull(assertion.getNotOnOrAfter());
+        assertTrue(assertion.isPresentlyValid());
     }
 
     @Test
     public void testSampleAssertion() throws Exception {
-        Element issuedAssertion = this.readDocument("/saml.xml").getDocumentElement();
-        String assertionId = issuedAssertion.getAttributeNodeNS(null, "ID").getNodeValue();
+        Element issuedAssertion = this.readDocument("/saml.xml")
+                .getDocumentElement();
+        String assertionId = issuedAssertion.getAttributeNodeNS(null, "ID")
+                .getNodeValue();
         SecurityToken token = new SecurityToken(assertionId, issuedAssertion, null);
         SecurityAssertionImpl assertion = new SecurityAssertionImpl(token);
         assertNotNull(assertion.getSecurityToken());
         assertEquals(token, assertion.getSecurityToken());
         assertEquals(ISSUER, assertion.getIssuer());
-        assertEquals(PRINCIPAL, assertion.getPrincipal().getName());
-        assertEquals(PRINCIPAL, assertion.getPrincipal().toString());
-        assertEquals(NUM_ATTRIBUTES, assertion.getAttributeStatements().size());
-        assertEquals(NUM_NAUTH, assertion.getAuthnStatements().size());
-        assertEquals(DatatypeConverter.parseDateTime(BEFORE).getTimeInMillis(),
-                assertion.getNotBefore().getTime());
-        assertEquals(DatatypeConverter.parseDateTime(AFTER).getTimeInMillis(),
-                assertion.getNotOnOrAfter().getTime());
+        assertEquals(PRINCIPAL,
+                assertion.getPrincipal()
+                        .getName());
+        assertEquals(PRINCIPAL,
+                assertion.getPrincipal()
+                        .toString());
+        assertEquals(NUM_ATTRIBUTES,
+                assertion.getAttributeStatements()
+                        .size());
+        assertEquals(NUM_NAUTH,
+                assertion.getAuthnStatements()
+                        .size());
+        assertEquals(DatatypeConverter.parseDateTime(BEFORE)
+                        .getTimeInMillis(),
+                assertion.getNotBefore()
+                        .getTime());
+        assertEquals(DatatypeConverter.parseDateTime(AFTER)
+                        .getTimeInMillis(),
+                assertion.getNotOnOrAfter()
+                        .getTime());
         //we don't currently parse these
         //        assertEquals(NUM_AUTHZ, assertion.getAuthzDecisionStatements().size());
         assertNotNull(assertion.toString());
+        assertTrue(assertion.isPresentlyValid());
+    }
+
+    @Test
+    public void testIsPresentlyValidWithNullBounds() throws Exception {
+        Element issuedAssertion = this.readDocument("/saml.xml")
+                .getDocumentElement();
+
+        //Remove Time Bounds
+        issuedAssertion.getElementsByTagName("saml2:Conditions")
```

Comment:
There are some places in the test where code is duplicated or nearly duplicated. You might want to consolidate some it in helper methods.